media/src/androidMain/kotlin/dev/icerock/moko/media/picker/ImagePickerDelegate.kt

```diff
@@ -0,0 +1,200 @@
+package dev.icerock.moko.media.picker
+
+import android.content.Context
+import android.net.Uri
+import androidx.activity.ComponentActivity
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.ActivityResultRegistryOwner
+import androidx.activity.result.PickVisualMediaRequest
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import dev.icerock.moko.media.BitmapUtils
+import kotlinx.coroutines.flow.MutableStateFlow
+
+internal class ImagePickerDelegate {
+
+    private var callback: CallbackData? = null
+
+    private val takePictureLauncherHolder = MutableStateFlow<ActivityResultLauncher<Uri>?>(null)
+    private val pickVisualMediaLauncherHolder =
+        MutableStateFlow<ActivityResultLauncher<PickVisualMediaRequest>?>(null)
```

Comment:
looks very simillar in multiple places. can we do reused implementation that will use in each case?